### ISSUE TITLE
Prettier handling of generic function types is broken

### ISSUE THREAD

**Issue by cvolant** [IanVS/prettier-plugin-sort-imports#8]
**Your Environment**

-   **Prettier version**: tried from 2.4.1 to 2.6.2
-   **node version**: 14.17.0
-   **package manager**: yarn@3.1.1
-   **IDE**: VScode

**Describe the bug**

I get errors with generic function types in `.ts` files: `const fn = <T>() => {}`.
```
Single type parameter T should have a trailing comma. Example usage: <T,>.
```
Then, if I add the trailing coma, Prettier removes it.

Trivago's original plugin works well, I get this problem only with this fork.
It may be related to Next.js babel config...

**To Reproduce**

```
yarn create-next-app@latest --typescript
yarn add @ianvs/prettier-plugin-sort-imports
```
Then reate a .ts file with the following declaration: `const fn = <T>() => {}`.

**Comment by IanVS** [comment]
Thanks, can you also share your prettier config?

**Comment by blutorange** [comment]
Hmm yeah, that's strange since we did not change anything related to parsing. If I take a look at the commits at https://github.com/trivago/prettier-plugin-sort-imports/commits/master, there also don't seem to be any additional commits since we created the fork. So I guess we need to take a look at what causes this different. And yes, a prettier config would be helpful.

**Comment by IanVS** [comment]
I'm not able to reproduce this in my own project.  Would you be willing to share a minimal reproduction, @cvolant?

**Comment by blutorange** [comment]
Also when I use the prettier online formatter, it removes the comma after type arguments.

Related:
* https://github.com/babel/babel/issues/14361
* https://stackoverflow.com/questions/71524043/single-type-parameter-t-should-have-a-trailing-comma-example-usage-t

![image](https://user-images.githubusercontent.com/2456413/165317592-5a658b4a-8ffb-426c-94fa-f023c4051690.png)

**Comment by IanVS** [comment]
It seems like maybe this is a babel error? https://github.com/babel/babel/issues/14361

**Comment by cvolant** [comment]
@IanVS @blutorange thank you for your answers, and sorry for the delay before mine : I was off for some time.
I followed the babel lead and finally succeeded to make it work: it was a indeed a babel issue.
For some reason, @babel/parser was resolved to version 7.17.7. I added the following to my package.json to force it to resolve to version 7.17.10 and it worked (I use Yarn) :
```json
"resolutions": {
  "@babel/parser": "7.17.10"
}
```
